lib/rules/indent.js

```diff
@@ -165,6 +165,20 @@ module.exports = {
             MemberExpression: null
         };
 
+        /*
+         * General rule strategy:
+         * 1. Create a hashmap (`desiredOffsets`). The keys are all the tokens and comments in the file, and the values are objects
+         *    containing information for a specific offset, measured in indent levels, from a either a specific token or the first column.
+         *    For example, an element in an array will have {offset: 1, from: openingCurly} to indicate that it is offset by one indentation
```

Comment:
Thank you for writing this! That said, you might want to switch "array" to "object" or "curly" to "square bracket".